### ISSUE TITLE
Invalid syntax in generated version.py 

### ISSUE THREAD

**Issue by ghost** [pypa/setuptools-scm#103]
A package I use is distributed through Mercurial and uses setuptools_scm with setup.py like this

``` python
from setuptools import setup
setup(
    use_scm_version={
        'write_to': 'PACKAGENAME/version.py',
    },
    setup_requires=['setuptools_scm'],
#...
)
```

The resulting version.py is not valid python:

``` python
# coding: utf-8
# file generated by setuptools_scm
# don't change, don't track in version control
version = <ScmVersion 0.4.0 d=None n=None d=False x={}>
```

The issue did not occur before today, it seems that it is caused by v1.12.0 release.

**Comment by untitaker** [comment]
Same issue here.

**Comment by RonnyPfannschmidt** [comment]
yikes, in response i pulled the release of pypi, i am now working on a fixup

**Comment by untitaker** [comment]
BTW this reordering creates the issue: https://github.com/pypa/setuptools_scm/commit/80565e3364f86f33be05d7d07be427f480f10d6a#diff-cc3eb8241d67f6487179a3c8976e128eR107

`format_version`'s rv is not used for `dump_version`.

**Comment by RonnyPfannschmidt** [comment]
i am aware, i'm testing the fix and i started Hungarian notation (parsed_version vs version_string) plus type assertions

**Comment by anarcat** [comment]
ouch, sorry - my bad. i assumed that since tests passed, things were okay. looks like the test suite was incomplete!

**Comment by RonnyPfannschmidt** [comment]
@anarcat @untitaker please take a look at the pr

**Comment by untitaker** [comment]
Fix lgtm. No tests though :P

**Comment by RonnyPfannschmidt** [comment]
i#ll fix in a different manner

**Comment by RonnyPfannschmidt** [comment]
the result of parse will always be a parsed version now

**Comment by RonnyPfannschmidt** [comment]
fixed in #104
